src/Step/ConverterStep.php

```diff
@@ -29,6 +29,8 @@ public function __construct(array $converters = [])
 
     /**
      * @param callable $converter
+     *
+     * @return self
```

Comment:
Well, as I know both is accepted. AFAIK @frankdejonge asked a PHPDoc developer once about this. The reason I prefer `self` is that it is a keyword which means the same class, while `$this` is not.

Another thing while I try to always use `self` is that it is always appropriate. For example an immutable class shouldn't return `$this` because it is not returning the same instance.

At least this is what I know about documenting fluent interface, but I admit this is a bit vague.